Cypress

mgabor18 opened the pull request:
Add cypress solution.

mgabor18 committed:
Setup project for Cypress

- npm init
- install cypress
- update gitignore

mgabor18 committed:
Add cypress tests and helpers

sorosz89 commented on cypress/fixtures/data.json:
```diff
@@ -0,0 +1,142 @@
+{
```
Would it make sense to isolate test data and configuration-specific data?

mgabor18 replied:
Separated test data.

sorosz89 commented on package.json:
```diff
@@ -0,0 +1,20 @@
+    "test:cypress:request-validation": "cypress run --spec 'cypress/interceptors/request-validation.spec.js'",
+    "test:cypress:request-modification": "cypress run --spec 'cypress/interceptors/request-modification.spec.js'"
+  },
+  "author": "Various Authors",
```
You can add your name here, or we can do it in the final version

mgabor18 replied:
I planned to do it separately, as the package.json will be shared

sorosz89 commented on cypress/interceptors/response-validation.spec.js:
```diff
@@ -0,0 +1,66 @@
+const { assertSchema } = require('@cypress/schema-tools');
```
In general, it would be beneficial to add some UI-side verification to the test to represent the e2e flow we can do with interceptors. What do you think?

mgabor18 replied:
This is a great topic to discuss in our upcoming session, will proceed according to that.

mgabor18 committed:
Separate test data and update tests

mgabor18 committed:
Update cypress.json and tests

mgabor18 committed:
Migrate to TypeScript

- update tests & scripts
- update gitignore
- update package.json
- add tsconfig.json

mgabor18 committed:
Update response-validation.spec.ts

mgabor18 committed:
Update response-modification.spec.ts

mgabor18 committed:
Merge branch 'main' into cypress

mgabor18 committed:
Migrate to yarn and fix small test error
